spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -140,6 +148,14 @@ private Spark3Util.CatalogAndIdentifier catalogAndIdentifier(CaseInsensitiveStri
       selector = AT_TIMESTAMP + asOfTimestamp;
     }
 
+    if (branch != null) {
+      selector = BRANCH + branch;
```

Comment:
These names are used in other places for patterns, but here they are prefixes. I think it would help to name these `BRANCH_PREFIX` and `TAG_PREFIX`.